src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -834,7 +926,85 @@ NVTX.@annotate function set_diagnostic_edmf_precomputed_quantities_do_integral!(
+                   microphysics_model isa Microphysics2Moment
+
+                # 2M warm rain formation
+                compute_warm_precipitation_sources_2M!(
+                    Sᵖ_prev_level,
+                    Sᵖ_2_prev_level,
+                    S_n_liqʲ_prev_level,
+                    S_n_raiʲ_prev_level,
+                    S_q_liqʲ_prev_level,
+                    S_q_raiʲ_prev_level,
+                    ρʲ_prev_level,
+                    n_liqʲ_prev_level,
+                    n_raiʲ_prev_level,
+                    q_totʲ_prev_level,
+                    q_liqʲ_prev_level,
+                    q_iceʲ_prev_level,
+                    q_raiʲ_prev_level,
+                    q_snoʲ_prev_level,
+                    tsʲ_prev_level,
+                    dt,
+                    microphysics_2m_params,
+                    thermo_params,
+                )
+                # Snow not implemented
+                @. S_q_iceʲ_prev_level = FT(0)
+                @. S_q_snoʲ_prev_level = FT(0)
 
+                # Cloud sources from the updrafts
+                @. S_q_liqʲ_prev_level += cloud_sources(
+                    cloud_params.liquid,
+                    thermo_params,
+                    q_totʲ_prev_level,
+                    q_liqʲ_prev_level,
+                    q_iceʲ_prev_level,
+                    q_raiʲ_prev_level,
+                    q_snoʲ_prev_level,
+                    ρʲ_prev_level,
+                    TD.air_temperature(thermo_params, tsʲ_prev_level),
+                    dt,
+                )
+                @. S_q_iceʲ_prev_level += cloud_sources(
+                    cloud_params.ice,
+                    thermo_params,
+                    q_totʲ_prev_level,
+                    q_liqʲ_prev_level,
+                    q_iceʲ_prev_level,
+                    q_raiʲ_prev_level,
+                    q_snoʲ_prev_level,
+                    ρʲ_prev_level,
+                    TD.air_temperature(thermo_params, tsʲ_prev_level),
+                    dt,
+                )
+                # Aerosol activation
+                # TODO - right now I'm hijacking the
+                seasalt_num_prev_level = aerosol_params.c₀_seasalt
+                seasalt_mean_radius_prev_level = aerosol_params.α_seasalt
+                sulfate_num_prev_level = FT(0)
+
+                @. S_n_liqʲ_prev_level += aerosol_activation_sources(
+                    seasalt_num_prev_level,
+                    seasalt_mean_radius_prev_level,
+                    sulfate_num_prev_level,
+                    q_totʲ_prev_level,
+                    q_liqʲ_prev_level + q_raiʲ_prev_level,
+                    q_iceʲ_prev_level + q_snoʲ_prev_level,
+                    n_liqʲ_prev_level + n_raiʲ_prev_level,
+                    ρʲ_prev_level,
+                    max(
+                        0,
+                        w_component.(
+                            Geometry.WVector.(u³ʲ_data_prev_halflevel)
```

Comment:
I think this is correct! We need to pass the velocity component along gravity and I think this is it.